### PR TITLE
Feature/beneficiary image descriptions

### PR DESCRIPTION
**Pull request by RedVeil**

### CODE REVIEW

File: packages/app/components/Proposals/Form/ControlledTextInput.tsx [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
 
 interface ControlledTextInputProps {
   inputValue: string;
   id: string;
   placeholder: string;
   errorMessage: string;
-  updateInput: (value: string) => void;
   isValid: (input: string) => boolean;
-  navigation: Navigation;
+  updateInput?: (value: string) => void;
```
Comment (anthonymartin) [inline packages/app/components/Proposals/Form/ControlledTextInput.tsx:9]: the interface here is a bit more complicated than it was before.

the interface would be simplified if the `updateInput` argument  was modified like this:
```
updateInput: (value: string | {value: string, index: number }) => void;
```
then the inputIndex and updateIndexedInput could be removed altogether

File: packages/app/components/Proposals/Form/IpfsUpload.tsx [hunk 2]
```diff
@@ -82,25 +86,23 @@ function uploadMultipleImagesToPinata(files, localState, setLocalState) {
...
-  setLocalState:
-    | React.Dispatch<React.SetStateAction<string>>
-    | React.Dispatch<React.SetStateAction<string[]>>;
   imageDescription: string;
   imageInstructions: string;
   fileType: string;
   numMaxFiles: number;
-  navigation: Navigation;
+  setLocalState?: (input: string) => void;
+  setLocalStateMultiple?: (input: string[]) => void;
```
Comment (anthonymartin) [inline packages/app/components/Proposals/Form/IpfsUpload.tsx:94]: the extra `setLocalStateMultiple` prop could be avoided by updating the setLocalState prop as:

`setLocalState: (input: string | string[]) => void;`

**Review by anthonymartin — CHANGES_REQUESTED** [review]

File: packages/app/components/Proposals/Form/IpfsUpload.tsx (hunk 2 above)
Comment (RedVeil) [reply packages/app/components/Proposals/Form/IpfsUpload.tsx:94]: thats great! didnt know you could do this

File: packages/app/components/Proposals/Form/ControlledTextInput.tsx (hunk 1 above)
Comment (RedVeil) [reply packages/app/components/Proposals/Form/ControlledTextInput.tsx:9]: i set the index parameter of the function to optional and it worked. But dont i still need the inputIndex?

**Review by anthonymartin — APPROVED** [review]

**Review by mefuru — APPROVED** [review]

**Pull request merged**